File: Software University/04) PHP Web Development Basics/12) OOP FUNDAMENTALS - Exercises/04. Person/Person.php
<?php

namespace Van\Classes\Fourth;


class Person
{
    /** @var  string */
    protected $name;

    /** @var  int */
    protected $age;

    public function __construct(string $name, int $age)
    {
        $this->setName($name)
            ->setAge($age);
    }

    protected function getName(): string
    {
        return $this->name;
    }

    protected function setName(string $name): Person
    {
        if (strlen($name) <= 3) {
            throw new \Exception('Name\'s length should not be less than 3 symbols');
        }

        $this->name = $name;

        return $this;
    }

    protected function getAge(): int
    {
        return $this->age;
    }

    protected function setAge(int $age): Person
    {
        if ($age <= 0) {
            throw new \Exception('Age must be positive!');
        }

        $this->age = $age;

        return $this;
    }
}

File: Software_University/09_Java_Advanced_January_2019/14_Defining_Classes_Exercises/01_Opinion_Poll/Person.java
class Person {
    private int age;
    private String name;

    Person(int age, String name) {
        this.setAge(age);
        this.setName(name);
    }

    String getInfo() {
        return String.format("%s - %d", getName(), getAge());
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}